src/k8s/pkg/k8sd/controllers/feature.go

```diff
@@ -72,14 +72,19 @@ func NewFeatureController(opts FeatureControllerOpts) *FeatureController {
 func (c *FeatureController) Run(
 	ctx context.Context,
 	getClusterConfig func(context.Context) (types.ClusterConfig, error),
+	getLocalhostAddress func() (string, error),
 	notifyDNSChangedIP func(ctx context.Context, dnsIP string) error,
 	setFeatureStatus func(ctx context.Context, name types.FeatureName, featureStatus types.FeatureStatus) error,
 ) {
 	c.waitReady()
 	ctx = log.NewContext(ctx, log.FromContext(ctx).WithValues("controller", "feature"))
 
 	go c.reconcileLoop(ctx, getClusterConfig, setFeatureStatus, features.Network, c.triggerNetworkCh, c.reconciledNetworkCh, func(cfg types.ClusterConfig) (types.FeatureStatus, error) {
-		return features.Implementation.ApplyNetwork(ctx, c.snap, cfg.APIServer, cfg.Network, cfg.Annotations)
+		localhostAddress, err := getLocalhostAddress()
+		if err != nil {
+			return types.FeatureStatus{Enabled: cfg.Network.GetEnabled(), Message: "failed to determine the localhost address"}, fmt.Errorf("failed to get localhost address: %w", err)
```

Comment:
I think we should return `Enabled: false` in this case. According to previous (very old) discussions in the team, mostly with Angelos.

Reply:
Now that I'm thinking about it, the feature is "enabled", but not "working" or "responsive" or "ready". 
Maybe we need to change how we show this in the future, but for now, I think `Enabled: false` is probably the way to go.

Reply:
I've went with `Enabled: false` initially however what happens in the case that 
1. we already had a successful deployment of the helm chart
2. reconcile loop kicked in
3. localhost address determination failed
4. retry
AFAIK the deployment is still there, the feature is "enabled" technically right? 

I'm open to setting it to `false` just feels confusing tbh